documents/api.py

```diff
@@ -26,12 +27,32 @@ class AttachmentViewSet(ModelViewSet, NestedViewSetMixin):
 
 
 class DocumentViewSet(ModelViewSet):
-    queryset = Document.objects.all()
-    permission_classes = [HasServiceAPIKey | IsAdminUser]
     parser_classes = [MultiPartParser]
     serializer_class = DocumentSerializer
+    # Permission checking is done by the decorators on a method basis
+    permission_classes = [AllowAny]
+
+    def get_queryset(self):
+        if user := self.request.user:
+            # If the user is a superuser, return the whole set
+            if user.is_superuser:
+                return Document.objects.all()
+            # Otherwise, filter only the Documents for the user's service
+            elif service := self.request.service:
+                return Document.objects.filter(service=service)
+            # Finally, if there's no service on the request (e.g. a non-staff user),
+            # return the documents created by that user
+            else:
+                return Document.objects.filter(user=user)
```

Comment:
Not sure if we would also want to filter documents for non-staff users also based on the service i.e. when a non-staff/normal user is using this backend in the context of service x, they should only get documents for that service. But I guess we can also implement that as a query filter, so maybe this is fine as it is, at least for now.

Reply:
Hmmm good point. We're gonna need to be a bit more specific when filtering documents for staff/non-staff users, even though the permissions would kick in at that point anyway.

But hmmm... let's see what would make the most sense from an actual business point of view, maybe if there's a service specified somewhere, we filter just for that. Otherwise, we return all. 🤷